File: RoomInn/app/src/main/java/postpc/finalproject/RoomInn/ui/UnityHandler.java
package postpc.finalproject.RoomInn.ui;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.unity3d.player.UnityPlayer;
import com.unity3d.player.UnityPlayerActivity;

import postpc.finalproject.RoomInn.R;
import postpc.finalproject.RoomInn.models.RoomInnApplication;


public class UnityHandler extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_unity_handler);

        Button unityButton = findViewById(R.id.startUnityButton);
        unityButton.setOnClickListener(new View.OnClickListener() {
               @Override
               public void onClick(View view) {
                   Intent intent = new Intent(UnityHandler.this, UnityPlayerActivity.class);
                   startActivity(intent);
                   RoomInnApplication.getInstance().setPathToUnity(UnityPlayer.currentActivity.getExternalFilesDir("").getAbsolutePath());
               }
           }
        );
    }
}

File: RoomInn/app/src/main/java/postpc/finalproject/RoomInn/ui/UnityHandler.kt
package postpc.finalproject.RoomInn.ui

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import postpc.finalproject.RoomInn.R

class UnityHandler : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_unity_handler)
        val unityButton = findViewById<Button>(R.id.startUnityButton)
        unityButton.setOnClickListener {
            var intent = Intent(this@UnityHandler, MainUnityPlayerActivity::class.java)
            intent.putExtra("Scene Index", this.intent.getStringExtra("Scene Index"))
            intent.putExtra("Room Name", this.intent.getStringExtra("Room Name"))
            startActivity(intent)
//            if (sceneIndex == RoomUnityPlayerActivity.sceneIndex) {
//                UnityPlayer.UnitySendMessage("ScenceLoader", "loadScene",
//                    RoomUnityPlayerActivity.sceneIndex
//                )
//                intent = Intent(this@UnityHandler, RoomUnityPlayerActivity::class.java)
//                intent.putExtra("Room Name", this.intent.getStringExtra("Room Name"))
//            }
//            else {
//                intent = Intent(this@UnityHandler, ScanUnityPlayerActivity::class.java)
//            }
//            startActivity(intent)
//            getInstance().pathToUnity = UnityPlayer.currentActivity.getExternalFilesDir("")!!
//                .absolutePath
        }
    }
}